pkg/deployer/manifest/ensure.go

```diff
@@ -93,18 +95,66 @@ func (m *Manifest) Reconcile(ctx context.Context) error {
 		return err
 	}
 
-	m.DeployItem.Status.Phase = lsv1alpha1.ExecutionPhaseSucceeded
 	m.DeployItem.Status.ProviderStatus = statusData
+	if err := m.kubeClient.Status().Update(ctx, m.DeployItem); err != nil {
+		m.DeployItem.Status.LastError = lsv1alpha1helper.UpdatedError(m.DeployItem.Status.LastError,
+			currOp, "UpdateStatus", err.Error())
+		return err
+	}
+
+	return nil
+}
+
+func (m *Manifest) CheckResourcesHealth(ctx context.Context) error {
+	var (
+		currOp          = "CheckResourcesHealthManifests"
+		manifestDecoder = serializer.NewCodecFactory(ManifestScheme).UniversalDecoder()
+		status          = &manifest.ProviderStatus{}
+	)
+
+	if _, _, err := manifestDecoder.Decode(m.DeployItem.Status.ProviderStatus.Raw, nil, status); err != nil {
+		return err
+	}
+
+	if len(status.ManagedResources) == 0 {
+		return nil
+	}
+
+	objects := make([]*unstructured.Unstructured, len(status.ManagedResources))
+	for i, mr := range status.ManagedResources {
+		// do not check ignored resources.
+		if mr.Policy == manifest.IgnorePolicy {
+			continue
+		}
+		ref := mr.Resource
+		obj := kutil.ObjectFromTypedObjectReference(&ref)
+		objects[i] = obj
+	}
+
+	backoff := wait.Backoff{
+		Duration: 5 * time.Second,
+		Factor:   0,
+		Steps:    3,
+	}
```

Comment:
we should make the maximum timeout to wait configurable in the provider config.
Otherwise we could easily run into this timeout if some pods need a long time to start

Reply:
I was not sure how to proceed here so I thought it would be better to do 3 checks in 10s and return an error to allow the controller to take care of an other `DeployItems`.

Is it fine to wait `60s` for instance ?

In any case I will move this to a simple timeout and make it configurable.

Reply:
lets make it configurable and set the default to 60s.
We also have the retry from the deployitem so this is fine I think